### ISSUE TITLE
Circular JSON TypeError in @nestjs/axios

### ISSUE THREAD

**Issue by sahellebusch** [nestjs/nest#11303]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current behavior

Add HttpModule import to module, HttpService to class and receive the following error when running jest test:
```bash
 TypeError: Converting circular structure to JSON
        --> starting at object with constructor 'NestContainer'
        |     property 'modules' -> object with constructor 'Map'
        |     property 'InternalCoreModule' -> object with constructor 'Module'
        --- property 'container' closes the circle
        at stringify (<anonymous>)
```

### Minimum reproduction code

https://github.com/sahellebusch/nestjs-cats-example

### Steps to reproduce

1. `npm i`
2. `npm run test`

### Expected behavior

No circular type error.

### Package

- [ ] I don't know. Or some 3rd-party package
- [ ] <code>@nestjs/common</code>
- [ ] <code>@nestjs/core</code>
- [ ] <code>@nestjs/microservices</code>
- [ ] <code>@nestjs/platform-express</code>
- [ ] <code>@nestjs/platform-fastify</code>
- [ ] <code>@nestjs/platform-socket.io</code>
- [ ] <code>@nestjs/platform-ws</code>
- [ ] <code>@nestjs/testing</code>
- [ ] <code>@nestjs/websockets</code>
- [X] Other (see below)

### Other package

@nestjs/axios

### NestJS version

9.3.0

### Packages versions

See example repo.

### Node.js version

v16.15.0

### In which operating systems have you tested?

- [X] macOS
- [ ] Windows
- [ ] Linux

**Comment by jmcdo29** [comment]
Looks like an error when we're trying to throw an error about an improper test configuration (e.g. a missing provider). Interestingly enough, I only see this error when running multiple test files, not when I run each one individually.

**Comment by micalevisk** [comment]
I found that this isn't related with `@nestjs/axios`:

![image](https://user-images.githubusercontent.com/13461315/226402756-11d5c2fb-6c9f-4aae-90d8-bac6d239c196.png)

![image](https://user-images.githubusercontent.com/13461315/226402664-a0f7bd90-0f96-4943-8784-1c039cc17002.png)

**Comment by micalevisk** [comment]
and looks that's only happens in v9.3.9 (core, common and testing packages)

Not sure why https://github.com/nestjs/nest/compare/v9.3.9...v9.3.10

**Comment by micalevisk** [comment]
@kamilmysliwiec it happens on v9.3.10 and v9.3.11 (the latest version as of now). v9.3.9 went fine.

**Comment by micalevisk** [comment]
might be related with this issue on Jest side: https://github.com/facebook/jest/issues/10577 because it works as expected when using `detectOpenHandles: true`

**Comment by kamilmysliwiec** [comment]
Fixed in 9.3.12
